sentry-android-core/src/main/java/io/sentry/android/core/gestures/SentryGestureListener.java

```diff
@@ -0,0 +1,238 @@
+package io.sentry.android.core.gestures;
+
+import android.view.GestureDetector;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.Window;
+import io.sentry.Breadcrumb;
+import io.sentry.IHub;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.SentryAndroidOptions;
+import java.lang.ref.WeakReference;
+import java.util.Collections;
+import java.util.Map;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class SentryGestureListener implements GestureDetector.OnGestureListener {
+
+  private final @NotNull WeakReference<Window> windowRef;
+  private final @NotNull IHub hub;
+  private final @NotNull SentryAndroidOptions options;
+
+  private final ScrollState scrollState = new ScrollState();
+
+  public SentryGestureListener(
+    final @NotNull WeakReference<Window> windowRef,
+    final @NotNull IHub hub,
+    final @NotNull SentryAndroidOptions options
+  ) {
+    this.windowRef = windowRef;
+    this.hub = hub;
+    this.options = options;
+  }
+
+  public void onUp(final @NotNull MotionEvent motionEvent) {
+    final View decorView = ensureWindowDecorView("onUp");
+    final View scrollTarget = scrollState.targetRef.get();
+    if (decorView == null || scrollTarget == null) {
+      return;
+    }
+
+    if (scrollState.type == null) {
+      options
+        .getLogger()
+        .log(SentryLevel.INFO, "Unable to define scroll type. No breadcrumb captured.");
+      return;
+    }
+
+    final String direction = scrollState.calculateDirection(motionEvent);
+    addBreadcrumb(scrollTarget, scrollState.type,
+      Collections.singletonMap("direction", direction));
+    scrollState.reset();
+  }
+
+  @Override
+  public boolean onDown(final @Nullable MotionEvent motionEvent) {
+    if (motionEvent == null) {
+      return false;
+    }
+    scrollState.reset();
+    scrollState.startX = motionEvent.getX();
+    scrollState.startY = motionEvent.getY();
+    return false;
+  }
+
+  @Override
+  public boolean onSingleTapUp(final @Nullable MotionEvent motionEvent) {
+    final View decorView = ensureWindowDecorView("onSingleTapUp");
+    if (decorView == null || motionEvent == null) {
+      return false;
```

Comment:
should this fallback to the parent implementation instead?

```suggestion
      return super.onSingleTapUp(motionEvent);
```

What does the return type do here?

Reply:
There's no parent since it's an interface that we are implementing, so there's no `super` to delegate to. The return type should indicate whether we consumed the event or not, so other listeners in the downstream chain will/will not be called depending on that